src/main/java/org/templegalaxia/patterns/acjs/Spoken.java

```diff
@@ -0,0 +1,216 @@
+import heronarts.lx.modulator.SinLFO;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.templegalaxia.patterns.TemplePattern;
+
+/*
+ * “Do you not know that a man is not dead while his name is still spoken?”
+ *         -- Terry Pratchett, Going Postal
+ */
+
+@LXCategory("Acjs")
+public class Spoken extends TemplePattern {
+  private static final String NAME_DATA = "/Users/anton/Projects/Galaxia/Galaxia/assets/names.txt";
+  private static final int DASH_MULTIPLIER = 3;
+
+  public final CompoundParameter density =
+      new CompoundParameter("Lights", 1.0, 0, colors.length)
+          .setDescription("Number of lights to use (density).");
+
+  public final CompoundParameter birthRate =
+      new CompoundParameter("Rate", 1.0, 0, 1)
+          .setDescription("How likely a new name is to appear (per second).");
+
+  public final CompoundParameter dot =
+      (CompoundParameter)
+          new CompoundParameter("Dot speed", 1000, 0, 10000)
+              .setUnits(LXParameter.Units.MILLISECONDS)
+              .setDescription("Length of a 'dot' in morse code.");
+
+  public final CompoundParameter dotVariance =
+      (CompoundParameter)
+          new CompoundParameter("Dot variance", 0, 0, 1000)
+              .setUnits(LXParameter.Units.MILLISECONDS)
+              .setDescription("How variable dot length will be between names.");
+
+  private final HashMap<Character, String> morseLookup = new HashMap<>();
+  private final List<String> names = new ArrayList<>();
+  private final List<MorsePulser> pulsers = new ArrayList<>();
+  private final List<Integer> availableLights = new ArrayList<>();
+
+  private final Random rand = new Random();
+
+  public Spoken(LX lx) {
+    super(lx);
+    addParameter(density);
+    addParameter(birthRate);
+    addParameter(dot);
+    addParameter(dotVariance);
+
+    loadMorse();
+
+    for (int i = 0; i < colors.length; i++) availableLights.add(i);
+  }
+
+  void loadMorse() {
+    // Shamelessly boosted from the model load code.
+    CSVParser parser;
+    try {
+      parser =
+          new CSVParser(
+              new BufferedReader(
+                  new InputStreamReader(Spoken.class.getResourceAsStream("morse.csv"))),
+              CSVFormat.EXCEL.withHeader());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    for (CSVRecord csvRecord : parser) {
+      morseLookup.put(csvRecord.get("Letter").charAt(0), csvRecord.get("Morse"));
+    }
+  }
+
+  void loadNames() {
+    Path dataPath = Paths.get(NAME_DATA);
```

Comment:
See: https://processing.org/reference/loadStrings_.html

The absolute file path here isn't going to load for other people, relativize that and use the Processing helper to get a String[] in a single call.